src/modules/thirdParty/sagas.ts

```diff
@@ -41,13 +57,47 @@ export function* getContractInstance(
   return contractInstance
 }
 
+export function* getPublishItemsSignature(thirdPartyId: string, qty: number) {
+  const maticChainId: ChainId = yield call(getChainIdByNetwork, Network.MATIC)
+  const provider = new ethers.providers.Web3Provider((window as any).ethereum)
```

Comment:
shouldn't this provider come from decentraland-dapps? (instead of using window.ethereum)

Reply:
I tried! but seems like it's not exposing the `getSigner` method 🤔 
![image](https://user-images.githubusercontent.com/8763687/157441107-223130ae-abed-4dfb-a222-14a2127cc3a4.png)

Reply:
Hm true, what if we use this: https://github.com/decentraland/decentraland-transactions/blob/master/src/utils.ts#L29 ?

with `getConnectedProvider`

The reason why is that getConnectedProvider will correctly assign the provider, even if the user is doing something weird (like wallet connect for example)

Reply:
Updated to use `getConnectedProvider` and `eth_signTypedData_v4` as suggested!